VTUL/dlp-access
# Upgrades app to React v18 and Amplify v5

**Upgrades app to React v18 and Amplify v5.**
* * *

**JIRA Ticket**: (https://webapps.es.vt.edu/jira/browse/LIBTD-2670) (:star:)

* Other Relevant Links (Meeting note, project page, related pull requests, etc.)

# What does this Pull Request do? (:star:)
Upgrades app to React v18 and Amplify v5. Upgrades the majority of other packages as well because of the dependency spiral. Refactors code to address to breaking/recommended changes. 

# What's the changes? (:star:)
A in-depth description of the changes made by this PR. Technical details and possible side effects.

* Upgrades React and React/* packages to v18
* Upgrades Amplify and Amplify/* packages to v5
* Upgrades Material UI v4 to MUI v5
* Upgrades bootstrap to v5
* Upgrades PDFJS to v2 and refactors to render PDFs in a canvas element instead of an iframe, which is extremely rad
* **NOTE: Mirador is still technically pinned to React 16 but seems functional with the upgrade to 18. This will, however, require us to use the `--force` flag on any npm command that will result in packages being installed which is super lame**
* **ALSO NOTE: Several styling rules seem to have broken in the upgrade (seemingly all font related?). I'm assuming this is due to changing classnames in the bootstrap and/or MUI upgrades. I'll have to address these in a future PR to avoid having _too_ many changes in one commit.**

# How should this be tested?

* Review the generated preview here: https://upgrade.d2ysrrdhih4bgc.amplifyapp.com/ - OR - deploy the `upgrade` branch locally and pull the `dlpdev` Amplify backend: `amplify pull --appId d2ysrrdhih4bgc --envName dlpdev` 
* NOTE: if deploying locally you should delete the `node_modules` directory and the `package-lock.json` file. And then don't forget the `--force` flag when running `npm install --force` to reinstall everything.
* Check that the site still functions as expected, with the exception of a few font issues that will be addressed in a later PR.

# Additional Notes:
Any additional information that you think would be helpful when reviewing this PR.

* What branch to be used for testing this PR? `upgrade`
* Does this change require documentation to be updated? `yeah, probably`
* Does this change add any new dependencies? `oh yeah`
* Does this change require any other modifications to be made to the repository? `yes, we will need to use the "--force" flag on any npm command that will result in packages being installed until Mirador is updated.`
* Could this change impact execution of existing code? `yes, several things had to be refactored and future changes will need to conform to upgrade changes`

# Interested parties
Tag (@ mention) interested parties
@andreaWaldren 

(:star:) Required fields
